src/carma-streets/plot_message_frequencies.py

```diff
@@ -0,0 +1,103 @@
+import argparse
+from pathlib import Path
+import sys
+from parse_kafka_logs import KafkaLogMessageType
+
+from matplotlib import pyplot as plt
+from matplotlib import axes
+import pandas as pd
+
+def plot_message_frequencies(csv_dir: str, plots_dir: str, simulation: bool = False):
+    """Function to plot all message frequencies in a single figure
+
+    Args:
+        csv_dir (str): directory that holds CSV message data.
+        plots_dir (str): directory where create figure will be saved
+        simulation (bool,optional): bool flag to indicate whether data was collected in simulation. Defaults to False.
+    """
+    csv_dir_path = Path(csv_dir)
+    plots_dir_path = Path(plots_dir)
+    if csv_dir_path.is_dir():
+        if plots_dir_path.is_dir():
+            print(f"WARNING: {plots_dir} already exists. Contents will be overwritten.")
+        plots_dir_path.mkdir(exist_ok=True)
+        # Read CSV data
+        message_data = dict([])
+        time_sync_data = None
+        for csv_file in csv_dir_path.glob("*.csv"):
+            print(f'Reading csv file {csv_file.name} ...')
+            if csv_file.name == 'time_sync.csv':
+                time_sync_data = pd.read_csv(csv_file)
+            else:
+                df = pd.read_csv(csv_file)
+                message_data[csv_file.name.split('.')[0]] = df
+        fig, plots = plt.subplots(len(message_data), sharex=True, layout="constrained")
+        # Add simulation time to message data
+        for idx,( message_name, message_data_frame) in enumerate(message_data.items()):
```

Comment:
Consider using `zip()` to avoid using indices. Index-based iteration is discouraged in Python:

```python
for msg_plot, (msg_name, msg_df) in zip(plots, message_data.items()):
    # ...
```